### PR TITLE
add azure_rm_roleassginment_facts

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

add new module for azure_rm_roleassignment_facts, test pending on role definition merge.

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yungezz** [comment]
manual test output for reference:

## playbook code snippets
```
    - name: Create a role assignment (Check Mode)
      azure_rm_roleassignment:
        scope: "/subscriptions/{{ subscription_id }}/resourceGroups/{{ resource_group }}"
        assignee_object_id: "{{ principal_id }}"
        role_definition_id: "{{ role_definition_id }}"
      check_mode: yes
      register: output

    - name: Assert creating role definition check mode
      assert:
        that:
          - output.changed

    - name: Create a role assignment
      azure_rm_roleassignment:
        scope: "/subscriptions/{{ subscription_id }}/resourceGroups/{{ resource_group }}"
        assignee_object_id: "{{ principal_id }}"
        role_definition_id: "{{ role_definition_id }}"
      register: output

    - name: Assert creating role definition
      assert:
        that:
          - output.changed

    - name: Get facts
      azure_rm_roleassignment_facts:
        scope: "/subscriptions/{{ subscription_id }}/resourceGroups/{{ resource_group }}"
        assignee: "{{ principal_id }}"
      register: facts
    - name: dump output
      debug:
        var: facts

    - name: delete  role assignment
      azure_rm_roleassignment:
        name: 3d01af82-13d7-46be-a544-fec82b3ea0c3
        scope: "/subscriptions/{{ subscription_id }}/resourceGroups/{{ resource_group }}"

```

## output 
```
TASK [Create a role assignment (Check Mode)] *************************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:22
Tuesday 26 February 2019  15:18:39 +0800 (0:00:02.928)       0:00:04.390 ******
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: yungez
<127.0.0.1> EXEC /bin/sh -c 'echo ~yungez && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084 `" && echo ansible-tmp-1551165519.5372853-181544841564084="` echo /hoe/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084 `" ) && sleep 0'
Using module file /home/yungez/.ansible/roles/azure_preview_modules/library/azure_rm_roleassignment.py
<127.0.0.1> PUT /home/yungez/.ansible/tmp/ansible-local-115065llns2kn/tmpntlqykg4 TO /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084/AnsiballZ_azure_rm_roleassignment.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084/ /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084/AnsiballZ_azure_m_roleassignment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/yungez/src/ansible/venv/bin/python /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084/AnsiballZ_azure_rm_roleassignment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/yungez/.ansible/tmp/ansible-tmp-1551165519.5372853-181544841564084/ > /dev/null 2>&1 && sleep 0'
 [WARNING]: Azure API profile latest does not define an entry for AuthorizationManagementClient

changed: [localhost] => changed=true
  id: null
  invocation:
    module_args:
      ad_user: null
      adfs_authority_url: null
      api_profile: latest
      assignee_object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      auth_source: null
      cert_validation_mode: null
      client_id: null
      cloud_environment: AzureCloud
      name: null
      password: null
      profile: null
      role_definition_id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/Microsoft.Authorization/roleDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1
      secret: null
      state: present
      subscription_id: null
      tenant: null

TASK [Assert creating role definition check mode] ********************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:30
Tuesday 26 February 2019  15:18:42 +0800 (0:00:02.820)       0:00:07.210 ******
ok: [localhost] => changed=false
  msg: All assertions passed

TASK [Create a role assignment] **************************************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:35
Tuesday 26 February 2019  15:18:42 +0800 (0:00:00.054)       0:00:07.264 ******
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: yungez
<127.0.0.1> EXEC /bin/sh -c 'echo ~yungez && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526 `" && echo ansible-tmp-1551165522.4120975-183105516536526="` echo /hoe/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526 `" ) && sleep 0'
Using module file /home/yungez/.ansible/roles/azure_preview_modules/library/azure_rm_roleassignment.py
<127.0.0.1> PUT /home/yungez/.ansible/tmp/ansible-local-115065llns2kn/tmpam9audih TO /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526/AnsiballZ_azure_rm_roleassignment.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526/ /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526/AnsiballZ_azure_m_roleassignment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/yungez/src/ansible/venv/bin/python /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526/AnsiballZ_azure_rm_roleassignment.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/yungez/.ansible/tmp/ansible-tmp-1551165522.4120975-183105516536526/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => changed=true
  id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1/providers/Microsoft.Authorization/roleAssignments/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
  invocation:
    module_args:
      ad_user: null
      adfs_authority_url: null
      api_profile: latest
      assignee_object_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      auth_source: null
      cert_validation_mode: null
      client_id: null
      cloud_environment: AzureCloud
      name: null
      password: null
      profile: null
      role_definition_id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/Microsoft.Authorization/roleDefinitions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1
      secret: null
      state: present
      subscription_id: null
      tenant: null

TASK [Assert creating role definition] *******************************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:42
Tuesday 26 February 2019  15:18:52 +0800 (0:00:10.209)       0:00:17.474 ******
ok: [localhost] => changed=false
  msg: All assertions passed


TASK [Get facts] *****************************************************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:47
Tuesday 26 February 2019  15:30:24 +0800 (0:00:02.354)       0:00:03.826 ******
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: yungez
<127.0.0.1> EXEC /bin/sh -c 'echo ~yungez && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021 `" && echo ansible-tmp-1551166224.8838375-91137391833021="` echo /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021 `" ) && sleep 0'
Using module file /home/yungez/.ansible/roles/azure_preview_modules/library/azure_rm_roleassignment_facts.py
<127.0.0.1> PUT /home/yungez/.ansible/tmp/ansible-local-11869_mhhievq/tmp0rqd1zns TO /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021/AnsiballZ_azure_rm_roleassignment_facts.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021/ /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021/AnsiballZ_azure_rm_roleassignment_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/yungez/src/ansible/venv/bin/python /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021/AnsiballZ_azure_rm_roleassignment_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/yungez/.ansible/tmp/ansible-tmp-1551166224.8838375-91137391833021/ > /dev/null 2>&1 && sleep 0'
 [WARNING]: Azure API profile latest does not define an entry for AuthorizationManagementClient

ok: [localhost] => changed=false
  invocation:
    module_args:
      ad_user: null
      adfs_authority_url: null
      api_profile: latest
      assignee: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      auth_source: null
      cert_validation_mode: null
      client_id: null
      cloud_environment: AzureCloud
      name: null
      password: null
      profile: null
      scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1
      secret: null
      subscription_id: null
      tenant: null
  roleassignments:
  - id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1/providers/Microsoft.Authorization/roleAssignments/xxxxxxx-13d7-46be-a544-fec82b3ea0c3
    name: xxxxxxx-13d7-46be-a544-fec82b3ea0c3
    principal_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
    role_definition_id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/Microsoft.Authorization/roleDefinitions/xxxxxxx-592e-4ea0-a6e1-52e0449ecc2d
    scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1
    type: Microsoft.Authorization/roleAssignments

TASK [dump output] ***************************************************************************************************************************************************************************************
task path: /home/yungez/src/ansible-testapp/roleassignment.yml:52
Tuesday 26 February 2019  15:30:27 +0800 (0:00:02.888)       0:00:06.714 ******
ok: [localhost] =>
  facts:
    changed: false
    failed: false
    roleassignments:
    - id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1/providers/Microsoft.Authorization/roleAssignments/xxxxxxx-13d7-46be-a544-fec82b3ea0c3
      name: xxxxxxx-13d7-46be-a544-fec82b3ea0c3
      principal_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
      role_definition_id: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/providers/Microsoft.Authorization/roleDefinitions/xxxxxxx-592e-4ea0-a6e1-52e0449ecc2d
      scope: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/ansible-roledefinition1
      type: Microsoft.Authorization/roleAssignments
    warnings:
    - Azure API profile latest does not define an entry for AuthorizationManagementClient

```

File: lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, (@yungezz)
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py]: 2019

File: lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+module: azure_rm_roleassignment_facts
+version_added: "2.8"
+short_description: Gets Azure Role Assignment facts.
+description:
+    - Gets facts of Azure Role Assignment.
+
+options:
+    scope:
+        description:
+            - The scope of role assignment applies to.
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py]: The scope **that** role assignment applies to

File: lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            - For example, use /subscriptions/{subscription-id}/ for a subscription,
+            - /subscriptions/{subscription-id}/resourceGroups/{resourcegroup-name} for a resource group,
+            - /subscriptions/{subscription-id}/resourceGroups/{resourcegroup-name}/providers/{resource-provider}/{resource-type}/{resource-name} for a resource
+    name:
+        description:
+            - Name of role assignment.
+            - Mutual exclusive with I(assignee).
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py:34]: Mutually

File: lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+        if self.name:
+            self.results['roleassignments'] = self.get_by_name()
+        elif self.assignee:
+            self.results['roleassignments'] = self.get_by_assignee()
+        elif self.resource_group:
+            self.results['roleassignments'] = self.list_by_resource_group()
+        elif self.scope:
+            self.results['roleassignments'] = self.list_by_scope()
+
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_roleassignment_facts.py:179]: shouldn't module fail when no option is specified?

**Review by zikalino — APPROVED** [review]
in general nothing major.

**Comment by Fred-sun** [comment]
@yuwzho @nitzmahone @jborean93 review_need

**Comment by Fred-sun** [comment]
@yuwzho Please review and push for forward this PR when you're available! Thanks!

**Review by zikalino — APPROVED** [review]

**Pull request merged**